### ISSUE TITLE
Some VR Titles no longer marked as such on Linux

### ISSUE THREAD

**Issue by Matombo** [ValveSoftware/steam-for-linux#6892]
#### Your system information

* Steam client version (build number or date): Feb 11 2020 19:27:00
* Distribution (e.g. Ubuntu): Manjaro
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
A lot of vr games are no longer marked as vr on Linux
For example The Gallery - Episode 2 Heart of the Emberstone (The Gallery - Episode 1 Call of the Starseed is however marked as vr) or Serious sam fusion 2017 (beta) even though that game is one of the few native linux vr titles.
(the games still have the store tag for vr but if you select display only vr titles in big picture or a dynamic colletction they are no longer listed. Also the vr icon on the lower right of the banner in the steam library is missing)

**Comment by Matombo** [comment]
The issue went away after a complete reinstall of steam including deleting .local/share/Steam and all external library folders.

I moved games around in my external library folders and deleted some  app_manifests on they way that seemed to be of uninstalled games. Maybe that caused the issue.

**Comment by Matombo** [comment]
Nope the problem just showed up again, this time i was not moving any games to different folders.

**Comment by Matombo** [comment]
After some testing i can reliable reproduce the issue:
If one of the affected games is installed and steam is restarted i loses it marking as a vr game.
If it's uninstalled and steam is restarted it id again correctly shown as a vr game.
A more complete list of my vr games:

Games affected by the bug:
- Aparture Hand Lab
- Dance Collider
- The Gallery - Episode 2: Heart of the Emberstone
- Heart of the Emberstone: Coliseum
- Moss
- Pistol Whip
- VRChat
- Witching Tower

Games not affected by the bug:
- BallisticNG
- Everspace
- The Gallery - Episode 1: Call of the Starseed
- Keep Talking and nobody explodes
- The Lab
- Serious Sam VR: The last hope
- SUPERHOT VR
- The Talos Principle VR (The online game on this list that supports vr on linux nativly, all other run through proton)

**Comment by Lyle-Tafoya** [comment]
-  Steam client version (build number or date): Feb 15 2020 01:17:48
-  Distribution (e.g. Ubuntu): Arch Linux
-  Opted into Steam client beta?: Yes
-  Have you checked for system updates?: Yes

I have reproduced this issue with both the beta and non-beta version of the Steam client as well as the beta and non-beta version of SteamVR. I'm also suffering from this problem. It is quite severe to the point where it is completely impractical to do any VR gaming at the moment. 

I want to clarify what "marked as vr on Linux" means in the context of the original report. Assuming I am suffering from the same issue as Matombo (I'm pretty sure I am), the games will still show up in the "VR" category in the desktop client. Where they do not show up is inside SteamVR when you try to select a game to play. The overlay that shows you all your installed VR games will simply be empty of any games. If you install a game, the newly installed game will show up in the list of installed VR games. If you restart the client, any/all games that were in that list before will no longer appear until you uninstall the game, restart the client again, .

However, this issue is far more severe than that. If that were the only issue, it wouldn't be a huge deal. You would simply be able to search the non-VR games and launch it from there or launch it with your mouse from the desktop client. However, this does not work either. In fact, games which are impacted by this issue will completely (and silently) fail to launch. The only way to play the game again is to uninstall it, restart Steam, and then reinstall the game. Just to clarify; you have to reinstall any given VR game and restart Steam EVERY TIME you want to play it (unless you keep Steam open all the time). This makes VR gaming incredibly tedious. This is made even worse by the fact that most VR gaming is done via Proton and many of these games run via Proton store save files inside the wine prefix. This means that I have to backup save files for some games if I want to reinstall them without losing my progress (ie. my favorite VR game "Distance"). So now every time I start Steam, I have to backup a save file, uninstall the game I want to play, close steam, start steam, install the game, and restore the save file before I can play the game. That is a whole lot of nonsense to have to go through every time I want to play a game.

@kisak-valve Is there a relevant dev we can ping to have look at this sooner rather than later? This makes VR games borderline unplayable.

**Comment by codingcatgirl** [comment]
This affects Beat Saber as well and i can confirm it does not affect Superhot VR.

**Comment by codingcatgirl** [comment]
I can also confirm that reinstalling the game fixes the issue. I don't even have to restart steam. However, as soon as i restart steam, the issue is back and Beat Saber vanishes from the list of VR games again. Fun!

**Comment by lachlan-aa** [comment]
Steam client version (build number or date): Feb 11 2020 19:27:00
Distribution: Manjaro
Opted into Steam client beta?: No
Have you checked for system updates?: Yes

I can confirm that this issue is still present now and occurs on every beta of steam vr. The only game in my library that isn't affected by this is SUPERHOT VR and the following games are affected by this bug: Blade & Sorcery, Budget Cuts, GORN, Pavlov VR and VR Chat.
Also there seems to be a correlation between the Steam VR library and the bug, as SUPERHOT VR (the only game in my library unaffected) is the only game that shows up in my VR games section in the Vr Big Picture mode.

Running Steam through the terminal reveals that all of the affected games output the same error when launched (no errors would show up when launching these games normally, it would just stop launching):
For Blade & Sorcery:
------------------------------------------------------------------------------------------------------------------------------------
Opted-in Controller Mask for AppId 629730: 0
CAPIJobRequestUserStats - Server response failed 2
GameAction [AppID 629730, ActionID 2] : LaunchApp changed task to ShowLaunchOption with ""
GameAction [AppID 629730, ActionID 2] : LaunchApp waiting for user response to ShowLaunchOption ""
GameAction [AppID 629730, ActionID 2] : LaunchApp continues with user response "0"
GameAction [AppID 629730, ActionID 2] : LaunchApp changed task to BPMHybridOverlay10ftShutdown with ""
Loaded Config for Local Override Path for App ID 629730, Controller 0: /home/lachlana/.local/share/Steam//controller_base/empty.vdf
GameAction [AppID 629730, ActionID 2] : LaunchApp changed task to ProcessingInstallScript with ""
esync: up and running.
esync: up and running.
esync: up and running.
GameAction [AppID 629730, ActionID 2] : LaunchApp changed task to SynchronizingCloud with ""
GameAction [AppID 629730, ActionID 2] : LaunchApp failed with WaitingOpenVRAppQuit with "104"
GameAction [AppID 629730, ActionID 2] : LaunchApp changed task to Failed with ""
-----------------------------------------------------------------------------------------------------------------------------------
I found a strange fix for this though, and it's easier than reinstalling the games. If you launch them without opening Steam VR first, the game will launch and open Steam VR on its own. This is only a work-around though and this issue needs to be addressed.

**Comment by codingcatgirl** [comment]
The fix of starting the game without opening Steam VR first does not work with games where that has always been an issue too, for example Beat Saber.

**Comment by Matombo** [comment]
I don't think this issue is related to games not starting, i have games that are still marked as vr games but arn't starting anyway.
I made another issue for the not starting games: https://github.com/ValveSoftware/steam-for-linux/issues/6896

**Comment by Zamundaaa** [comment]
I'm also affected by some games not showing up in the Dashboard, notably Beat Saber. I'm not entirely sure other games are affected for me, gonna have a look once I'm home. I can't launch Beat Saber as well, so the two issues may very well be not the same but sitll correlated.

**Comment by Lyle-Tafoya** [comment]
@Zamundaaa  Not being able to launch the games is definitely part of the issue

**Comment by Zamundaaa** [comment]
I just had Steam in offline mode (internet problems) and Beat Saber appeared in the dashboard and was launchable. It's still there in online mode now as well.

**Comment by Matombo** [comment]
Can confirm that both this bug and https://github.com/ValveSoftware/steam-for-linux/issues/6896 are not present in offline mode.

However for me they reapear as soon as i go online again.

**Comment by Baguettedood** [comment]
I decided to change Proton versions for Beat Saber away from the whitelisted version and noticed that after selecting "Force the use of a specific Steam Play compatibility tool", it became marked as VR again.

**Comment by Zamundaaa** [comment]
Are you opted into the Steam Beta? There's been a fix mentioned in the update notes.

**Comment by Baguettedood** [comment]
I did notice the beta partially fixes it, allowing SteamVR to run the titles, though they still aren't always marked as VR titles in the non-VR Steam library.

**Comment by Matombo** [comment]
https://github.com/ValveSoftware/steam-for-linux/issues/6896 is fixed for me too, but this bug is still present.

**Comment by farmboy0** [comment]
I have a new behaviour with the steam beta. VR games start only in desktop mode. Ive tried Budget Cuts, Cosmic Trip, Apex Construct Demo and Superhot VR. This might be related to this bug although they are still shown in Steam VR.

**Comment by kisak-valve** [comment]
Hello, is anyone still experiencing this issue on an up to date system?

**Comment by Baguettedood** [comment]
I re-examined my manually-created VR games collection and it looks like most of the games I manually added now also show up in the "VR" filter.

The only exceptions I could find were:

- Payday 2 (Windows version has VR but I have the Linux version installed)
- Tabletop Simulator (Windows version has VR but I have the Linux version installed)
- Nekopalive (Might be a misconfiguration specific to this game)

I had *We Were Here* manually added though I think its VR support was removed.

Other than that it appears to be fixed.

**Comment by farmboy0** [comment]
I have created a dynamic collection with only hardware VR support selected.
With two 2 games manually added i have 207 titles in the list.
I see the following games with VR support(checked each of them against the store page) still in uncategorized:
* Explore Fushimi Inari
* Fingers: Mini Games
* Intruders: Hide and Seek
* Neos VR
* PowerBeatsVR - VR Fitness
* Transference™
* Vacation Simulator
* The Vanishing of Ethan Carter VR
* Zaccaria Pinball
